src/task_4_2_get_ranges.py

```diff
@@ -0,0 +1,55 @@
+"""Task_4_2."""
```

Comment:
Overall it looks as overhead
it's task for technical interview. 
it seems practically impossible to implement such solution in 10-15 minutes and it in fact doesn't have any advantages.